python/lsst/daf/butler/butler.py

```diff
@@ -69,104 +69,65 @@ def __init__(self, config):
         if self.run is None:
             self.run = self.registry.makeRun(self.config['run'])
 
-    def getDirect(self, ref, parameters=None):
-        """Load a `Dataset` or a slice thereof from a `DatasetRef`.
-
-        Unlike `Butler.get`, this method allows `Datasets` outside the Butler's `Collection` to be read as
-        long as the `DatasetRef` that identifies them can be obtained separately.
+    def put(self, obj, datasetType, dataId, producer=None):
+        """Store and register a dataset.
 
         Parameters
         ----------
-        ref : `DatasetRef`
-            A pointer to the `Dataset` to load.
-        parameters : `dict`
-            `StorageClass`-specific parameters that can be used to obtain a slice of the `Dataset`.
+        obj : `object`
+            The dataset.
+        datasetType : `DatasetType` instance or `str`
+            The `DatasetType`.
+        dataId : `dict`
+            An identifier with `DataUnit` names and values.
```

Comment:
Are you referring to the class here because the keys are names that convert to instances of `DataUnit` classes?

Reply:
No. They are names of the tables. (e.g. `{"camera" : "HSC", "visit" : 3}`).